runtime/autoload/health.vim

```diff
@@ -1,141 +1,135 @@
-function! s:trim(s) abort
-  return substitute(a:s, '^\_s*\|\_s*$', '', 'g')
-endfunction
 
+" Dictionary where we keep all of the healtch check functions we've found.
+" They will only be run if they are true
+let g:health_checkers = get(g:, 'health_checkers', {})
+let s:current_checker = get(s:, 'current_checker', '')
 
-" Simple version comparison.
-function! s:version_cmp(a, b) abort
-  let a = split(a:a, '\.')
-  let b = split(a:b, '\.')
+function! health#check(bang) abort
+  let l:report = 'Checking health'
 
-  for i in range(len(a))
-    if a[i] > b[i]
-      return 1
-    elseif a[i] < b[i]
-      return -1
-    endif
-  endfor
 
-  return 0
-endfunction
+  for l:checker in items(g:health_checkers)
+    " Disabled checkers will not run their registered check functions
+    if l:checker[1]
+      let s:current_checker = l:checker[0]
+      let l:report .= printf("\nChecker %s says: %s", s:current_checker, l:checker[1])
 
+      let l:report .= capture('call ' . l:checker[0] . '()')
```

Comment:
Thanks!  Then, it would only make sense to use `try` if we want a checker to have an exit code (I can't think of a reason where we would).  Otherwise, I think `capture('silent! foo')` would be the best way to go.